frontend/src/pages/RunList.tsx

```diff
@@ -429,14 +432,27 @@ class RunList extends React.PureComponent<RunListProps, RunListState> {
 
   /**
    * For each run ID, fetch its corresponding run, and set it in DisplayRuns
+   * Remove the run from the list if its name does not match the filter
    */
-  private _getAndSetRuns(displayRuns: DisplayRun[]): Promise<DisplayRun[]> {
+  private _getAndSetRuns(displayRuns: DisplayRun[], filter: ApiFilter): Promise<DisplayRun[]> {
+    const filterSubstring =
+      filter.predicates &&
+      filter.predicates[0] &&
+      filter.predicates[0].key === 'name' &&
+      filter.predicates[0].op === PredicateOp.ISSUBSTRING &&
+      filter.predicates[0].string_value
+        ? filter.predicates[0].string_value.toLowerCase()
+        : '';
     return Promise.all(
       displayRuns.map(async displayRun => {
         let getRunResponse: ApiRunDetail;
         try {
           getRunResponse = await Apis.runServiceApi.getRun(displayRun.run!.id!);
-          displayRun.run = getRunResponse.run!;
+          if (getRunResponse.run?.name?.toLowerCase().includes(filterSubstring)) {
+            displayRun.run = getRunResponse.run!;
+          } else {
+            displayRuns.splice(displayRuns.indexOf(displayRun), 1);
```

Comment:
This may be an unconventional approach to filter out unwanted elements (typically a new array with only the desired elements would be created), but I wanted to maintain both the orders of the runs and the parallel asynchronous `getRun` requests, and thought this outweighed the larger `O(n)` delay for removing each element. This is especially because the `runIdListMask` means that the array will likely be small (in the case of the `Compare.tsx` page, a maximum of 10 runs). However, I can change this approach if the readability is poor.

Reply:
Resolving this comment as this code is now outdated, as we decided to use the `filter` function instead.